src/openforms/js/components/admin/form_design/registrations/zgw/fields/ProductSelect.js

```diff
@@ -0,0 +1,99 @@
+import {useFormikContext} from 'formik';
+import PropTypes from 'prop-types';
+import {FormattedMessage} from 'react-intl';
+import {components} from 'react-select';
+import useAsync from 'react-use/esm/useAsync';
+
+import Field from 'components/admin/forms/Field';
+import FormRow from 'components/admin/forms/FormRow';
+import ReactSelect from 'components/admin/forms/ReactSelect';
+import {get} from 'utils/fetch';
+
+const PRODUCTS_ENDPOINT = '/api/v2/registration/plugins/zgw-api/products';
+
+const getAvailableProducts = async (apiGroupID, catalogueUrl, caseTypeIdentification) => {
+  const response = await get(PRODUCTS_ENDPOINT, {
+    zgw_api_group: apiGroupID,
+    catalogue_url: catalogueUrl,
+    case_type_identification: caseTypeIdentification,
+  });
+  if (!response.ok) {
+    throw new Error('Loading available case type products failed');
+  }
+  return response.data.map(({uri, description}) => ({
+    value: uri,
+    label: uri,
+    description: description,
+  }));
+};
+
+const ProductOption = props => {
+  const {label, description} = props.data;
+  return (
+    <components.Option {...props}>
+      {description && <span className="catalogi-type-option">{description}</span>}
+      <code>{label}</code>
```

Comment:
I don't really see much benefit in displaying the full URI here. Instead, if there's no description, you can use the URI as fallback, but otherwise just showing the description is sufficient.

Having `getAvailableProducts` return `{value: uri, label: description || uri}` seems like the much simpler approach here, then you also don't need a custom option component.

Reply:
Mhh okay check. Yeah I thought it might be nice to see which product it precisely is..

Yeah if only the description or the url is shown, then that would be a better approach